src/analyses/mCPRegistry.ml

```diff
@@ -251,24 +273,41 @@ struct
       else fold_left (fun a (n,d) -> f a n d @@ assoc n y) a x
     with Not_found -> raise (DomListBroken "binop_fold : assoc failure")
 
+  let binop_short_fold_bool f a (x:t) (y:t) =
+    let f n d1 d2 =
+      f n (assoc_dom n) d1 d2
+    in
+    try if length x <> length y
+      then raise (DomListBroken "binop_short_fold_bool : differing lengths")
+      else (
+        iter (fun (n,d) -> f n d @@ assoc n y) x;
+        a
+      )
+    with
+      | Not_found -> raise (DomListBroken "binop_short_fold_bool : assoc failure")
+      | ShortCircuitBool b -> b
+
```

Comment:
Do we really need to give `binop_short_fold_bool` both in `DomListPrintable` and in `DomListLattice`?

Reply:
It's for the same reason there were `binop_fold` in both before: one uses first class modules of type `(module Printable.S)` and the other of `(module Lattice.S)`.